powerstore/resource_snapshotrule_test.go

```diff
@@ -241,6 +243,67 @@ func TestAccSnapshotRule_ReadSnapShotRuleUnavailable(t *testing.T) {
 	})
 }
 
+// Test to import resource but resulting in error
+func TestAccSnapshotRule_ImportFailure(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testProviderFactory,
+		Steps: []resource.TestStep{
+			{
+				Config:        SnapshotRuleParamsWithTimeOfDay,
+				ResourceName:  "powerstore_snapshotrule.test",
+				ImportState:   true,
+				ExpectError:   regexp.MustCompile("Could not import snapshot rule"),
+				ImportStateId: "invalid-id",
+			},
+		},
+	})
+}
+
+// Test to import successfully
+func TestAccSnapshotRule_ImportSuccess(t *testing.T) {
+
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testProviderFactory,
+		Steps: []resource.TestStep{
+			{
+				Config: SnapshotRuleParamsWithTimeOfDay,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "name", "test_snapshotrule"),
+					resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "time_of_day", "21:00"),
+					resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "timezone", "UTC"),
+					resource.TestCheckTypeSetElemAttr("powerstore_snapshotrule.test", "days_of_week.*", "Monday"),
+					resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "desired_retention", "56"),
+					resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "nas_access_type", "Snapshot"),
+					resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "is_read_only", "false"),
+				),
+			},
+			{
+				Config:       SnapshotRuleParamsWithTimeOfDay,
+				ResourceName: "powerstore_snapshotrule.test",
+				ImportState:  true,
```

Comment:
Please verify if  ImportStateVerify flag is needed as well?

Reply:
included